File: x-engine-module-localstorage/android/module-localstorage/src/main/java/com/zkty/modules/loaded/jsapi/__xengine__module_localstorage.java
package com.zkty.modules.loaded.jsapi;

import com.zkty.modules.engine.utils.ActivityUtils;
import com.zkty.modules.loaded.util.SharePreferenceUtils;

import com.zkty.modules.dsbridge.CompletionHandler;

public class __xengine__module_localstorage extends xengine__module_localstorage {


    @Override
    public void __testSetOtherIDLocalStorage(StorageSetDTO dto, CompletionHandler<StorageStatusDTO> handler) {

    }

    @Override
    public void __testGetOtherIDLocalStorage(StorageGetDTO dto, CompletionHandler<StorageStatusDTO> handler) {

    }

    @Override
    public void _setLocalStorage(StorageSetDTO dto, CompletionHandler<StorageStatusDTO> handler) {
        SharePreferenceUtils.put(ActivityUtils.getCurrentActivity(), dto.isPublic, dto.key, dto.value);
        StorageStatusDTO statusDTO = new StorageStatusDTO();
        statusDTO.result = "success";
        handler.complete(statusDTO);
    }

    @Override
    public void _getLocalStorage(StorageGetDTO dto, CompletionHandler<StorageStatusDTO> handler) {
        String result = (String) SharePreferenceUtils.get(ActivityUtils.getCurrentActivity(), dto.isPublic, dto.key, "");
        StorageStatusDTO statusDTO = new StorageStatusDTO();
        statusDTO.result = result;
        handler.complete(statusDTO);
    }


    @Override
    public void _removeLocalStorageAll(StorageRemoveDTO dto, CompletionHandler<StorageStatusDTO> handler) {
        SharePreferenceUtils.clear(ActivityUtils.getCurrentActivity(), dto.isPublic);
        StorageStatusDTO statusDTO = new StorageStatusDTO();
        statusDTO.result = "success";
        handler.complete(statusDTO);
    }

    @Override
    public void _removeLocalStorageItem(StorageRemoveDTO dto, CompletionHandler<StorageStatusDTO> handler) {
        SharePreferenceUtils.remove(ActivityUtils.getCurrentActivity(), dto.isPublic, dto.key);
        StorageStatusDTO statusDTO = new StorageStatusDTO();
        statusDTO.result = "success";
        handler.complete(statusDTO);
    }

}

File: modules/x-engine-module-localstorage/iOS/Class/__xengine__module_localStorage.m
//
//  __xengine__module_localstorage.m
//  AFNetworking
//
//  Created by Chenwuzheng on 2020/8/6.
//

#import "__xengine__module_localstorage.h"
 
#import "XEDataSaveManage.h"

@implementation __xengine__module_localstorage


- (void)_get:(StorageGetDTO *)dto complete:(void (^)(StorageStatusDTO *, BOOL))completionHandler {
    StorageStatusDTO *ret = [StorageStatusDTO new];
    ret.result = [XEDataSaveManage getLocalStorage:dto.key withIsPublic:dto.isPublic];
    completionHandler(ret , YES);
}

- (void)_remove:(StorageRemoveDTO *)dto complete:(void (^)(StorageStatusDTO *, BOOL))completionHandler {
    [XEDataSaveManage removeLocalStorageItem:dto.key withIsPublic:dto.isPublic];
    StorageStatusDTO *ret = [StorageStatusDTO new];
    ret.result = @"success";
    completionHandler(ret , YES);
}

- (void)_removeAll:(StorageRemoveAllDTO *)dto complete:(void (^)(StorageStatusDTO *, BOOL))completionHandler {
    [XEDataSaveManage removeLocalStorageAll:dto.isPublic];
    StorageStatusDTO *ret = [StorageStatusDTO new];
    ret.result = @"success";
    completionHandler(ret , YES);
}

- (void)_set:(StorageSetDTO *)dto complete:(void (^)(StorageStatusDTO *, BOOL))completionHandler {
    [XEDataSaveManage setLocalStorage:dto.key withValue:dto.value withIsPublic:dto.isPublic];
    StorageStatusDTO *ret = [StorageStatusDTO new];
    ret.result = @"success";
    completionHandler(ret , YES);
}

@end
